### PR TITLE
New PHPStorm OSX cheat sheet

### PR DESCRIPTION
**Pull request by juhosa**
**What does your Instant Answer do?**
It shows small portion of keyboard shortcuts for OSX version of PHPStorm. By small I mean the the most useful ones in my opinion.

**What is the data source for your Instant Answer? (Provide a link if possible)**
http://overapi.com/phpstorm

**Why did you choose this data source?**
The PDF is provided by JetBrains, but it looks like they don't host it anymore, so this link is one working that I found.

**Are there any other alternative (better) data sources?**
I gonna go with 'no'.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
PHP programmers.

**Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?**
To Cheat Sheet Goodie.

**Are you having any problems? Do you need our help with anything?**
Nope

**Where did you hear about DuckDuckHack? (For first time contributors)**
From DuckDuckGo twitter account.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
screen shot 2015-12-19 at 01 05 31

### CODE REVIEW

**Comment by daxtheduck** [comment]
@juhosa Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub ID:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata post coming soon

**Comment by jophab** [comment]
@juhosa LGTM :+1:

File: share/goodie/cheat_sheets/json/phpstorm-osx.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
+{
+    "id": "phpstorm_osx_cheat_sheet",
+    "name": "JetBrains PHPStorm IDE (OS X)",
+    "descriptions": "Keyboard shortcuts for OS X version PHPStorm IDE, a PHP IDE developed by JetBrains",
+    "metadata": {
+        "sourceName": "OverAPI",
+        "sourceUrl": "http://overapi.com/phpstorm/"
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/phpstorm-osx.json]: Can you change this sourceUrl to be `http://overapi.com/static/cs/PhpStorm_ReferenceCard.pdf`

File: share/goodie/cheat_sheets/json/phpstorm-osx.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            "val": "Optimize imports"
+        }, {
+            "key": "[Ctrl] [Opt] [I]",
+            "val": "Auto indent lines"
+        }],
+        "Searching/Replacing": [{
+            "key": "[Shift] [Shift]",
+            "val": "Search everywhere"
+        }, {
+            "key": "[⌘]  [F]",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/phpstorm-osx.json]: Nit: there's an extra space here between the `]` and `[`

File: share/goodie/cheat_sheets/json/phpstorm-osx.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            "val": "Find"
+        }, {
+            "key": "[⌘] [G]",
+            "val": "Find next"
+        }, {
+            "key": "[Shift] [⌘] [G]",
+            "val": "Find previous"
+        }, {
+            "key": "[⌘]  [R]",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/phpstorm-osx.json]: Same again on the spacing :)

File: share/goodie/cheat_sheets/json/phpstorm-osx.json [hunk 4]
```diff
@@ -0,0 +1,82 @@
+{
+    "id": "phpstorm_osx_cheat_sheet",
+    "name": "JetBrains PHPStorm IDE (OS X)",
+    "descriptions": "Keyboard shortcuts for OS X version PHPStorm IDE, a PHP IDE developed by JetBrains",
+    "metadata": {
+        "sourceName": "OverAPI",
+        "sourceUrl": "http://overapi.com/phpstorm/"
+    },
+    "aliases": ["phpstorm os x", "phpstorm mac"],
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/phpstorm-osx.json]: Can you add `phpstorm macos` and `phpstorm mac os` to the aliases?

**Comment by juhosa** [comment]
@mintsoft I've made the changes.

**Comment by mintsoft** [comment]
Many Thanks @juhosa  LGTM!

**Comment by moollaza** [comment]
@juhosa thanks for contributing!

@jophab @mintsoft thanks for merging!

We need to make sure an IA page is created first though, so the community platform is aware of this Instant Answer.

I'll make the IA page for this one :+1:

**Comment by moollaza** [comment]
This one is now live on beta: https://beta.duckduckgo.com/?q=phpstorm+osx+cheat+sheet&t=ffcm&ia=cheatsheet

**Comment by moollaza** [comment]
@juhosa congrats! It's live: https://duckduckgo.com/?q=phpstorm+osx+cheat+sheet&t=ffcm&ia=answer

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Comment by moollaza** [comment]
@juhosa congrats again on getting your Cheat Sheet merged. I wanted to get in touch with you to see if there’s anything I can help you with, or if you’d like me to suggest some more ways in which you can help out (bugs, new IA’s etc)?

I’m reaching out to our recent, successful contributors to see what brought them to DuckDuckHack, and what else you’d like to work on? Perhaps something more difficult like a full Goodie or API-Powered Spice Instant Answer?

Also if you have any feedback or thoughts on your experience contributing to DuckDuckHack I’d really appreciate it!

Cheers!

**Comment by juhosa** [comment]
@moollaza Sorry for the super late response!

I would definitely like to do something more, I just haven't got the time to do anything. Maybe in the summer when work doesn't keep me so busy :) I'm open to any suggestions though. (Plus, I love the T-shirt, so maybe a hoodie after like 5 merged PRs? :P)

Contributing to DuckDuckHack was fairly simple and made really accessible, because everything that is needed is documented.

**Comment by moollaza** [comment]
@juhosa hey there, thanks for getting back to me!

> I just haven't got the time to do anything

I totally understand that, everyone's lives are busy!

If you'd like, you can join our DuckDuckHack Contributors team to be ping when we need help and so you can be assigned tasks.

Also we have a developer newsletter which you can signup for to receive our weekly email.


> (Plus, I love the T-shirt, so maybe a hoodie after like 5 merged PRs? :P)

Hahaha I'm glad you like the t-shirt. We very recently got Hoodies for the Staff so hopefully those will be available to the public soon. Lots of my friends want one!

**Comment by tagawa** [comment]
@juhosa, thanks again for creating the PHPStorm OSX cheat sheet! You’ve done a really good job here and we'd like you to help our community tackle the next big challenge - our mission to become the best search engine for programmers: https://duck.co/blog/post/297/help-for-programmers.

When you're ready to move on, please: 

- Create a new Instant Answer to increase our Programming IA coverage

OR

- Test live programming Instant Answers (or those in development) for bugs and suggest improvements

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage).

Thanks again!

**Pull request merged**